pacai/bin/gridworld.py

```diff
@@ -168,12 +167,15 @@ def getTransitionStatesAndProbs(self, state, action):
         return successors
 
     def __aggregate(self, statesAndProbs):
-        counter = Counter()
+        counter = {}
         for state, prob in statesAndProbs:
-            counter[state] += prob
+            if state in counter:
+                counter[state] += prob
+            else:
+                counter[state] = prob
```

Comment:
You should check if the state isnt in counter, then initialize it to 0. After just add prob to it. This avoids an if else statement and brings it down into just an if statement.